### DISCUSSION TITLE
Is it possible to have transcluded blocks wrap the text

### DISCUSSION THREAD

**Discussion by dcoales** [zsviczian/obsidian-excalidraw-plugin#145] [Q&A] [answered]
I'm very much a visual thinker and like the idea of spatial tools for thought.  I've recently been looking at [project meta](https://projectmeta.app) and think it looks very exciting.  I can't help feeling though that I could set up much the same thing with QuickAdd and Excalidraw automate if I was a bit smarter.  

I just want a way to automate adding an existing note onto a drawing so that it displays as a white box with the note contents transcluded inside the box with the text wrapped so the box isn't too wide.  Perhaps not all the contents need to be transcluded - maybe just the first header (if there is one) and the first paragraph after the header.

I think that would provide most of the best bits of what I can see in the short demo video playing on the site.  

The main drawback I see at the moment though is that transcluded blocks appear as one long line rather than wrapping to some defined width.  

Do you think it's possible to do this kind of thing with Excalidraw now .... or will it become possible with features you have on the roadmap.

Any

**Comment by zsviczian** [comment, marked as answer]
Transcluded text can be wrapped. Try `![[doc#heading]]{40}` or `![[doc#^blockref]]{50}` where 40 and 50 are the line length.

If you want to automate with QuickAdd, you can use ExcalidrawAutomate.addText(). This function also supports text wrapping, as well as automatically adding a box around the text. https://zsviczian.github.io/obsidian-excalidraw-plugin/API/objects.html#addtext

**Comment by dcoales** [comment]
Thanks, I didn't know about the option to wrap to a fixed width.  I'll read the automate manual and see what I can do.

**Reply by mdroidian** [reply]
I'm very interested to see your progress here!
